src/features/content-explorer/content-explorer/ContentExplorer.js

```diff
@@ -181,17 +181,39 @@ class ContentExplorer extends Component {
     }
 
     componentDidUpdate({ initialFoldersPath: prevInitialFoldersPath }) {
-        const { initialFoldersPath } = this.props;
+        const { initialFoldersPath, isSelectAllAllowed } = this.props;
+        const { isSelectAllChecked } = this.state;
 
         if (prevInitialFoldersPath !== initialFoldersPath) {
             this.handleFoldersPathUpdated(initialFoldersPath);
         }
+        if (!this.isLoadingItems() && isSelectAllAllowed) {
```

Comment:
I have tested this locally, and once the items do actually load, the checkbox will update again. So, the checkbox and the items remain in sync during this phase. I added this if statement because it is a better user experience for the checkbox to not update until the new items load. Without the isLoading check, the checkbox unchecks itself whenever items are loading and then updates once they are loaded in. So navigating between folders with everything selected would lead to a checked box going from checked to unchecked then checked again, which I thought looked worse.